### PR TITLE
[DDW-405] Wait Daedalus to quit before launching app update

### PR DESCRIPTION
**Pull request by daniloprates**
This PR changes the way the update installer is launched, waiting first for the app to quit.

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Review by gnpf — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "daedalus",
   "productName": "Daedalus",
-  "version": "2.3.0-FC2",
+  "version": "2.1.0",
```
Comment (nikolaglumac) [inline package.json:4]: @daniloprates please revert this change. The version should remain 2.3.0-FC2...

File: source/main/ipc/manageAppUpdateChannel.js [hunk 2]
```diff
@@ -173,14 +173,10 @@ export const handleManageAppUpdateRequests = (window: BrowserWindow) => {
     // For linux we execute the installer file
     if (environment.isLinux) return installUpdate(filePath);
 
-    // For other OS we launch the installer file
-    const openInstaller: boolean = shell.openItem(filePath);
-    if (!openInstaller)
```
Comment (nikolaglumac) [inline source/main/ipc/manageAppUpdateChannel.js:178]: @daniloprates shouldn't we have this error case handled?

**Comment by ManusMcCole** [comment]
Hi @daniloprates. The functionality worked perfectly on Mainnet. Is that sufficient for testing purposes ? On Flight it gets stuck on "Trying to fetch the newsfeed " on builds 14590 and 14591 with these messages in the console see <a href="https://drive.google.com/file/d/1Mp4AkZqwzIwDI8GiNg01peqIt7uTAC-z/view?usp=sharing">screenshot</a>

**Review by mioriohk — APPROVED** [review]
Working as expected. Tested on build 14590 for both Windows and Mac. Great job @daniloprates 👍

File: source/main/ipc/manageAppUpdateChannel.js (hunk 2 above)
Comment (daniloprates) [reply source/main/ipc/manageAppUpdateChannel.js:178]: @nikolaglumac this is only for [testing purposes](https://github.com/input-output-hk/daedalus/pull/2195/commits/03ff1637d59f92e6107d86c8f09ed06373ea4552).

File: package.json (hunk 1 above)
Comment (daniloprates) [reply package.json:4]: @nikolaglumac this is only for [testing purposes](https://github.com/input-output-hk/daedalus/pull/2195/commits/03ff1637d59f92e6107d86c8f09ed06373ea4552).

**Comment by daniloprates** [comment]
> Hi @daniloprates. The functionality worked perfectly on Mainnet. Is that sufficient for testing purposes ? On Flight it gets stuck on "Trying to fetch the newsfeed " Win and Mac on builds 14590 and 14591 with these messages in the console see [screenshot](https://drive.google.com/file/d/1Mp4AkZqwzIwDI8GiNg01peqIt7uTAC-z/view?usp=sharing)

@ManusMcCole yes, it's due to the dummy data I've included, which is targeted to mainnet only.

File: source/main/ipc/manageAppUpdateChannel.js (hunk 2 above)
Comment (daniloprates) [reply source/main/ipc/manageAppUpdateChannel.js:178]: @nikolaglumac because we use `app.exit(exitCode)` instead of `app.quit()` the `will-quit` event is [not emitted](https://github.com/electron/electron/blob/v8.2.2/docs/api/app.md#appexitexitcode). This could be done only by heavily changing our quitting flow, which I don't think it's worth, since this error will probably never occur, only  if the file is corrupted. We check if the file exists before quitting Daedalus. Since we only rename the file to its final name after the download is completed, if the file is there, there's probably no chance for the error to happen.

**Review by ManusMcCole — APPROVED** [review]

File: source/main/ipc/manageAppUpdateChannel.js (hunk 2 above)
Comment (daniloprates) [reply source/main/ipc/manageAppUpdateChannel.js:178]: @nikolaglumac [testing data removed](https://github.com/input-output-hk/daedalus/pull/2195/commits/e50bd82664618d29fe7bb349cfe3ab6538b83a78). Let me know if you want me to go for this change ☝️ but I don't think it's necessary.

Comment (nikolaglumac) [reply source/main/ipc/manageAppUpdateChannel.js:178]: Sounds good @daniloprates 👍

**Review by nikolaglumac — APPROVED** [review]
Great work!
